App/Sagas/TextileSagas.js

```diff
@@ -239,25 +239,61 @@ export function * shareImage ({thread, hash, caption}) {
 
 export function * photosTask () {
   try {
-    const photos = yield call(queryPhotos)
-    for (const photo of photos.reverse()) {
-      const multipartData = yield call(IPFS.addImageAtPath, photo.path, photo.thumbPath, 'default')
-      yield call(RNFS.unlink, photo.path)
-      yield call(RNFS.unlink, photo.thumbPath)
-      yield put(TextileActions.imageAdded('default', multipartData.boundary, multipartData.payloadPath))
-      yield put(IpfsNodeActions.getPhotoHashesRequest('default'))
-      yield call(
-        Upload.startUpload,
-        {
-          customUploadId: multipartData.boundary,
-          path: multipartData.payloadPath,
-          url: 'https://ipfs.textile.io/api/v0/add?wrap-with-directory=true',
-          method: 'POST',
-          type: 'multipart',
-          field: multipartData.boundary
+    const camera = yield select(TextileSelectors.camera)
+    const processed = camera && camera.processed ? camera.processed : []
+
+    let allPhotos = yield call(getAllPhotos)
+    if (processed.length === 0) {
+      const ignoredPhotos = allPhotos.splice(1)
+      for (const photo of ignoredPhotos) {
+        yield put(TextileActions.imageIgnore(photo.uri))
+      }
+    }
+
+    let allProcessed = processed.reduce((o, item, index) => ({...o, [item]: { index }}), {})
+    const photos = allPhotos.filter((photo) => {
+      if (allProcessed[photo.uri]) {
+        return false
+      }
+      return true
+    })
+
+    for (const orig of photos.reverse()) {
+      let photo
+      try { // single photo try/catch
+        photo = yield call(scalePhoto, orig)
+      } catch (error) {
+        yield put(TextileActions.photoProcessingError(orig.uri, error))
+        continue // jump to next photo
+      }
+
+      // ** Only runs if photo scaling was successful **
```

Comment:
If you move the call to scale the photo into this `try`, the desired behavior described here will be true... If the photo scaling fails, all the subsequent code inside the `try` won't run, it will just call `catch` and `finally` and then control will go back to the `for` loop and the next iteration of the loop will run. I think that's right.

Reply:
the reason i did it this way is so that i was safe to call `call(RNFS.unlink, photo.path)` in the finally below. if i put this in the same try and it failed the call to finall unlink would also fail

Reply:
or i move the scale and then wrap the finally content in a new try, you see?

Reply:
You already have the `unlink` in the `finally` wrapped in another try/catch, right?

Reply:
ah, the looking closer, the real reason was thinking through what information in the error would actually be actionable. i can just change all errors to deal with uri as the identifier and then  yours works

Reply:
coming up

Reply:
cool